src/mdsynthesis/treants.py

```diff
@@ -73,55 +69,32 @@ def universe(self):
         """
         # TODO: include check for changes to universe definition, not just
         # definition absence
-        if self._universe:
-            return self._universe
+        # if self._universe:
+        #     return self._universe
+        # else:
+        args = self.universedef._args
+        kwargs = self.universedef.kwargs
+        if args is None:
+            self._universe = None
         else:
-            self.universedef._activate()
-            return self._universe
+            self._universe = mda.Universe(*args, **kwargs)
+        return self._universe
```

Comment:
@dotsdl  I assume you don't like this too much. Right now we reparse the universe everytime we call `sim.universe`. It's because I'm trying to make `UniverseDefinition` a class that only stores information and doesn't change anything. Before you had this tight coupling between the UniverseDefinition and the Sim object. That allowed neat things like.

```python
s.universedef.topology=other
s.universe # dynamically reload
s.universe # use cached version
```

That is pretty neat, to be honest. I'm not sure how to best solve that. My current idea would be to calculate a hash from the UniverseDefinition and only reload the universe when that hash has changed. That might take care of your TODO as well.